env/work/srw_python/srwl_uti_smp.py

```diff
@@ -0,0 +1,321 @@
+    :param shift_x: shift the whole image horizontally. Positive value shifts the image to the right, negative - to the left. See numpy.pad() for details.
+    :param shift_y: shift the whole image vertically. Positive value shifts the image to the top, negative - to the bottom. See numpy.pad() for details.
+    :param invert: invert the image. See numpy.invert() for details.
+    :param is_save_images: a flag to save the initial and processed images.
+    :param prefix: the prefix to add to the names of the saved image files.
+    :param output_image_format: the format of the output file. If not specified, the input format is used.
+    :return: transmission (SRWLOptT) type optical element which simulates the Sample.
+    """
+
+    input_parms = {
+        "type": "sample",
+        "resolution": resolution,
+        "thickness": thickness,
+        "refractiveIndex": delta,
+        "attenuationLength": atten_len,
+        "horizontalCenterCoordinate": xc,
+        "verticalCenterCoordinate": yc,
+        "initialPhotonEnergy": e_start,
+        "finalPhotonPnergy": e_fin,
+        'area': area,
+        'rotateAngle': rotate_angle,
+        'rotateReshape': rotate_reshape,
+        'cutoffBackgroundNoise': cutoff_background_noise,
+        'backgroundColor': background_color,
+        'tile': tile,
+        'shiftX': shift_x,
+        'shiftY': shift_y,
+        'invert': invert,
+        'outputImageFormat': output_image_format,
+    }
+
+    s = SRWLUtiSmp(
+        file_path=file_path,
+        area=area,
+        rotate_angle=rotate_angle,
+        rotate_reshape=rotate_reshape,
+        cutoff_background_noise=cutoff_background_noise,
+        background_color=background_color,
+        tile=tile,
+        shift_x=shift_x,
+        shift_y=shift_y,
+        invert=invert,
+        is_show_images=False,
+        is_save_images=is_save_images,
+        prefix=prefix,
+        output_image_format=output_image_format,
+    )
+
+    # Input parameters to SRWLOptT:
+    nx = s.nx
+    ny = s.ny
+    rx = nx * resolution
+    ry = ny * resolution
+
+    opT = srwlib.SRWLOptT(_nx=nx, _ny=ny, _rx=rx, _ry=ry,
+                          _arTr=arTr, _extTr=extTr, _Fx=fx, _Fy=fy,
+                          _x=xc, _y=yc, _ne=ne, _eStart=e_start, _eFin=e_fin)
+
+    data = s.data
+
+    # Same data alignment as for wavefront: outmost loop vs y, inmost loop vs e
+    offset = 0
+    for iy in range(ny):
+        for ix in range(nx):
+            for ie in range(ne):
+                # In images Y=0 corresponds from upper-left corner, in SRW it's lower-left corner:
+                pathInBody = thickness * data[ny - iy - 1, ix] / s.limit_value
+                opT.arTr[offset] = math.exp(-0.5 * pathInBody / atten_len)  # amplitude transmission
+                opT.arTr[offset + 1] = -delta * pathInBody  # optical path difference
+                offset += 2
+
+    opT.input_parms = input_parms
+
+    return opT
```

Comment:
OK, I took the entire file (without testing).

Reply:
I've tested it with Sirepo yesterday, seems to work fine.